### PR TITLE
[Lifecycle] start/stop ksensor onCreate and onDestroy

### PR DESCRIPTION
**Pull request by sukhwanyoon**
- added handleOnUndock in onDestroy
    - moved handleOnDock in OnDestroy
    - change return type of setMain in GVRActiviyDelegate

Signed-off-by: Sean Yoon <sukhwan.y@samsung.com>

### CODE REVIEW

**Comment by sukhwanyoon** [comment]
Hi. Mihail,
Please informed that it's only for my project for now, which means not for drop-in. Please give me an advice how to modify this so that this can work without bothering drop-in module.
Thanks.
Sean,

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRActivity.java [hunk 1]
```diff
@@ -100,7 +100,8 @@ protected void onCreate(Bundle savedInstanceState) {
                     mDelegate = (GVRActivityDelegate) aClass.newInstance();
                     mAppSettings = mDelegate.makeVrAppSettings();
                     mDelegate.onCreate(this);
-
+                    mActivityNative = mDelegate.getActivityNative();
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRActivity.java]: There is already code to invoke handleOnDock immediately if the backend requests it. GVRActivity.java:
```
    public void setMain(GVRMain gvrMain, String dataFileName) {
        this.mGVRMain = gvrMain;
        if (getRequestedOrientation() == ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE) {
            onConfigure(dataFileName);
            mDelegate.setMain(gvrMain, dataFileName);

            boolean isMonoscopicMode = mAppSettings.getMonoscopicModeParams().isMonoscopicMode();
            if (!isMonoscopicMode) {
                mViewManager = mDelegate.makeViewManager();
            } else {
                mViewManager = mDelegate.makeMonoscopicViewManager();
            }
            mDelegate.setViewManager(mViewManager);

            if (mConfigurationManager.isDockListenerRequired()) {
                startDockEventReceiver();
            } else {
                handleOnDock();
            }
```

So this is not needed.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRActivity.java [hunk 2]
```diff
@@ -218,10 +219,13 @@ protected void onDestroy() {
             mDockEventReceiver.stop();
         }
 
+        handleOnUndock();
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRActivity.java]: Calling onUndock is not there yet. @sivapenke has raised an issue about it. Will continue the discussion in the other repo.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRActivity.java [hunk 3]
```diff
@@ -242,8 +246,10 @@ protected void onDestroy() {
     public void setMain(GVRMain gvrMain, String dataFileName) {
         this.mGVRMain = gvrMain;
         if (getRequestedOrientation() == ActivityInfo.SCREEN_ORIENTATION_LANDSCAPE) {
+            if(false == mDelegate.setMain(gvrMain, dataFileName)) {
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRActivity.java]: We can take the change to have setMain return a result.

File: GVRf/Framework/framework/src/main/java/org/gearvrf/GVRActivity.java [hunk 4]
```diff
@@ -256,7 +262,8 @@ public void setMain(GVRMain gvrMain, String dataFileName) {
             if (mConfigurationManager.isDockListenerRequired()) {
                 startDockEventReceiver();
             } else {
-                handleOnDock();
+                //android.util.Log.w(TAG, "setMain calls handleOnDock decided by isDockListenerRequired");
+                //handleOnDock();
```
Comment (liaxim) [inline GVRf/Framework/framework/src/main/java/org/gearvrf/GVRActivity.java]: This is the earliest possible place where handleOnDock can be invoked and have the backend correctly initialized. Let me know in an email why this didn't work for you, thanks.

**Comment by liaxim** [comment]
Will create a pull request with the change where setMain returns a result.

**Comment by liaxim** [comment]
Please look at this pull request: https://github.com/Samsung/GearVRf/pull/1096

**Comment by liaxim** [comment]
@sukhwanyoon If you don't mind I plan to close this pull request.

**Pull request closed without merging**